RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletEDProducerT.cc

```diff
@@ -86,21 +86,21 @@ void CAHitNtupletEDProducerT<T_Generator>::produce(edm::Event& iEvent, const edm
   generator_.initEvent(iEvent, iSetup);
 
   LogDebug("CAHitNtupletEDProducer") << "Creating ntuplets for " << regionDoublets.regionSize() << " regions, and " << regionDoublets.layerPairsSize() << " layer pairs";
-
-  typename T_Generator::ResultType ntuplets;
-  ntuplets.reserve(localRA_.upper());
-
+  std::vector<typename T_Generator::ResultType> ntuplets;
+  ntuplets.resize(regionDoublets.regionSize());
+  for(auto ntuplet : ntuplets)  ntuplet.reserve(localRA_.upper()); 
+ 
+  generator_.hitNtuplets(regionDoublets, ntuplets, iSetup, seedingLayerHits);
```

Comment:
Eventually I'd prefer seedingHitSets (or similar) to be passed directly to generator to avoid temporary vectors (but to be done later).